packages/flutter_tools/lib/executable.dart

```diff
@@ -64,8 +65,9 @@ Future<void> main(List<String> args) async {
       (args.isNotEmpty && args.first == 'help') || (args.length == 1 && verbose);
   final bool muteCommandLogging = help || doctor;
   final bool verboseHelp = help && verbose;
+  final bool daemon = args.contains('daemon');
 
-  await runner.run(args, <FlutterCommand>[
+  await runner.run(args, () => <FlutterCommand>[
```

Comment:
Making creation lazy, so that we can setup the zone and inject feature flags/platforms to commands directly. This will let us avoid globals in the commands classes, and also gate commands like `build web` behind the right flag